rust/sdk-processor/src/config/indexer_processor_config.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::processor_config::ProcessorConfig;
+use crate::processors::events_processor::EventsProcessor;
+use anyhow::Result;
+use aptos_indexer_processor_sdk::aptos_indexer_transaction_stream::TransactionStreamConfig;
+use sdk_server_framework::RunnableConfig;
+use serde::{Deserialize, Serialize};
+
+pub const QUERY_DEFAULT_RETRIES: u32 = 5;
+pub const QUERY_DEFAULT_RETRY_DELAY_MS: u64 = 500;
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct IndexerProcessorConfig {
+    pub processor_config: ProcessorConfig,
+    pub transaction_stream_config: TransactionStreamConfig,
+    pub db_config: DbConfig,
```

Comment:
should this even be a top level config? theoretically the processor might not depend on the database, only the progress tracker. so probably make it a ProgressTrackerConfig?

Reply:
This is only used by our processors, hence why it's not in the aptos-indexer-processor-sdk crate. This IndexerProcessorConfig only serves as an example and you can add/remove fields as needed.

Reply:
should we expect most of external people still reuse whatever we provide, they may add processors, but whatever common/framework code we provide here will likely be reused.

even for us, from an interface design point of view, this should look like:

```
struct IndexerProcessorConfig {
  progress_tracker_config: ProgressTrackerConfig,
  processor_config: ...
  ...
}

struct ProgressTrackerConfig {
  db_config: PostgresDbConfig,  // we should probably even make it an enum here to allow using things other than pg here.
}

struct EventsProcessorConfig {
  event_store_config: EventsStoreConfig,
  ...
}

struct EventsStoreConfig {
    db_config: PostgresDbConfig,  // we should probably even make it an enum here to allow using things other than pg here.
}

struct ParquetDefaultProcessorConfig {
  // no db_config here
}

```

Reply:
I think we should keep things simple, without creating so many configs that all contain the same thing which makes deployment and code more complicated: 

```
struct IndexerProcessorConfig {
  db_config: Option<DbConfig>, // could be enum
  parquet_config: Option<ParquetConfig>,
  processor_config: ...
  ...
}
```